### PR TITLE
Add NRPE checks for services

### PR DESCRIPTION
**Pull request by xavpaice**
Adds the nrpe-external-master layer and checks for services managed by
this charm.

func-test-pr: https://github.com/openstack-charmers/zaza-openstack-tests/pull/433

Closes-Bug: #1896674

### CODE REVIEW

File: lib/charms/ovn_charm.py [hunk 1]
```diff
@@ -608,6 +610,24 @@ def configure_bridges(self):
...
+        hostname = nrpe.get_nagios_hostname()
+        current_unit = nrpe.get_nagios_unit_name()
+        # Determine if this is a subordinate unit or not
+        if ch_core.hookenv.principal_unit() == ch_core.hookenv.local_unit():
+            primary = True
+        else:
+            primary = False
+        charm_nrpe = nrpe.NRPE(hostname=hostname, primary=primary)
+        if reactive.is_flag_set('nova-compute.available'):
+            self.nrpe_check_services.append(self.metadata_agent)
```
Comment (fnordahl) [inline lib/charms/ovn_charm.py:627]: The charm class `__init__` method already makes choices based on OpenStack being present or not, could we do this there instead?

File: lib/charms/ovn_charm.py [hunk 2]
```diff
@@ -621,13 +641,18 @@ def __init__(self, **kwargs):
         """Allow augmenting behaviour on external factors."""
         super().__init__(**kwargs)
         if self.enable_openstack:
-            metadata_agent = 'networking-ovn-metadata-agent'
+            self.metadata_agent = 'networking-ovn-metadata-agent'
```
Comment (fnordahl) [inline lib/charms/ovn_charm.py:645]: If you move the definition of `self.nrpe_check_services` above this block you can append the metadata agent to it when OpenStack is enabled. Then you don't need to expose the `metadata_agent` variable as a class instance variable.

File: lib/charms/ovn_charm.py [hunk 3]
```diff
@@ -116,6 +116,7 @@ class BaseOVNChassisCharm(charms_openstack.charm.OpenStackCharm):
     enable_openstack = False
     bridges_key = 'bridge-interface-mappings'
     nrpe_check_services = []
+    metadata_agent = 'neutron-ovn-metadata-agent'
```
Comment (fnordahl) [inline lib/charms/ovn_charm.py:119]: See previous comment about handling this in the per-release class `__init__` method which would make this redundant.

**Review by fnordahl — CHANGES_REQUESTED** [review]
Thank you for your work on this, it looks pretty great and I have a couple of comments in-line

**Comment by xavpaice** [comment]
Thanks for that feedback - I've updated with a fresh commit for some updates and improved the unit test for nrpe as a result.

**Review by fnordahl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**